dep crush on status -dot

In case status can't output version see https://github.com/golang/dep/issues/664

dep status -dot panics:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x13b757e]

goroutine 1 [running]:
main.(*dotOutput).BasicLine(0xc420199ce0, 0xc4201b61c0)
	/Users/gregory/Documents/gospace/src/github.com/golang/dep/cmd/dep/status.go:177 +0x2e
main.runStatusAll(0xc42013b9c0, 0x1680c00, 0xc420199ce0, 0xc420135080, 0x1682280, 0xc42001df80, 0x0, 0x1)
	/Users/gregory/Documents/gospace/src/github.com/golang/dep/cmd/dep/status.go:359 +0x8a4
main.(*statusCommand).Run(0xc420134d20, 0xc42013d9c0, 0xc4200102f0, 0x0, 0x0, 0x0, 0x0)
	/Users/gregory/Documents/gospace/src/github.com/golang/dep/cmd/dep/status.go:218 +0x1f2
main.(*Config).Run(0xc42001cfc0, 0xc42001cfc0)
	/Users/gregory/Documents/gospace/src/github.com/golang/dep/cmd/dep/main.go:161 +0x8e3
main.main()
	/Users/gregory/Documents/gospace/src/github.com/golang/dep/cmd/dep/main.go:44 +0x253
```

--- a/cmd/dep/status.go
+++ b/cmd/dep/status.go
@@ -174,7 +174,11 @@ func (out *dotOutput) BasicFooter() {
 }
 
 func (out *dotOutput) BasicLine(bs *BasicStatus) {
-	out.g.createNode(bs.ProjectRoot, bs.Version.String(), bs.Children)
+	version := formatVersion(bs.Revision)
+	if bs.Version != nil {
+		version = formatVersion(bs.Version)
+	}
+	out.g.createNode(bs.ProjectRoot, version, bs.Children)
 }
 
 func (out *dotOutput) MissingHeader()                {}

--- a/cmd/dep/status_test.go
+++ b/cmd/dep/status_test.go
@@ -5,8 +5,12 @@
 package main
 
 import (
+	"bytes"
 	"testing"
 
+	"strings"
+
+	"github.com/golang/dep"
 	"github.com/golang/dep/internal/gps"
 )
 
@@ -23,6 +27,41 @@ func TestStatusFormatVersion(t *testing.T) {
 		str := formatVersion(version)
 		if str != expected {
 			t.Fatalf("expected '%v', got '%v'", expected, str)
+		}
+	}
+}
+
+func TestBasicLine(t *testing.T) {
+
+	project := dep.Project{}
+
+	var tests = []struct {
+		status   BasicStatus
+		expected string
+	}{
+		{BasicStatus{
+			Version:  nil,
+			Revision: gps.Revision("flooboofoobooo"),
+		}, `[label="\nflooboo"];`},
+		{BasicStatus{
+			Version:  gps.NewVersion("1.0.0"),
+			Revision: gps.Revision("flooboofoobooo"),
+		}, `[label="\n1.0.0"];`},
+	}
+
+	for _, test := range tests {
+		var buf bytes.Buffer
+
+		out := &dotOutput{
+			p: &project,
+			w: &buf,
+		}
+		out.BasicHeader()
+		out.BasicLine(&test.status)
+		out.BasicFooter()
+
+		if ok := strings.Contains(buf.String(), test.expected); !ok {
+			t.Fatalf("Did not find expected node label: \n\t(GOT) %v \n\t(WNT) %v", buf.String(), test.status)
 		}
 	}
 }

